src/fairing.rs

```diff
@@ -122,19 +122,60 @@ impl Fairing for Compression {
 ///
 /// rocket::build()
 ///     // ...
-///     .attach(CachedCompression::fairing(vec![".otf", "main.dart.js"]))
+///     .attach(CachedCompression {
+///       cached_paths: vec!["", "/", "/about", "/people", "/posts", "/events", "/groups"],
+///       cached_path_prefixes: vec!["/user/", "/g/", "/p/"],
+///       cached_path_suffixes: vec![".otf", "main.dart.js"]
+///     })
 ///     // ...
 ///     # ;
 ///
 /// ```
 pub struct CachedCompression {
-    pub cached_path_endings: Vec<&'static str>,
+    pub cached_paths: Vec<&'static str>,
+    pub cached_path_prefixes: Vec<&'static str>,
+    pub cached_path_suffixes: Vec<&'static str>,
+    pub excluded_path_prefixes: Vec<&'static str>,
 }
 
 impl CachedCompression {
-    pub fn fairing(cached_path_endings: Vec<&'static str>) -> CachedCompression {
+    /// Caches only the specific paths provided.
+    pub fn specific_fairing(cached_paths: Vec<&'static str>) -> CachedCompression {
         CachedCompression {
-            cached_path_endings,
+            cached_paths,
+            cached_path_prefixes: vec![],
+            cached_path_suffixes: vec![],
+            excluded_path_prefixes: vec![],
+        }
+    }
+
+    /// Caches all paths with the provided suffixes.
+    pub fn suffix_fairing(cached_path_suffixes: Vec<&'static str>) -> CachedCompression {
+        CachedCompression {
+            cached_path_suffixes,
+            cached_paths: vec![],
+            cached_path_prefixes: vec![],
+            excluded_path_prefixes: vec![],
+        }
+    }
+
+    /// Caches all paths with the provided suffixes.
+    pub fn prefix_fairing(cached_path_prefixes: Vec<&'static str>) -> CachedCompression {
+        CachedCompression {
+            cached_path_prefixes,
+            cached_paths: vec![],
+            cached_path_suffixes: vec![],
+            excluded_path_prefixes: vec![],
+        }
+    }
+
+    /// Caches compressed responses for all paths except those with the excluded prefixes.
+    pub fn exclusion_fairing(excluded_path_prefixes: Vec<&'static str>) -> CachedCompression {
```

Comment:
Similarly here, I think this name should be changed to something like `path_exclusion_fairing()` so users know what is being excluded more easily based on the name.